modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/java/com/liferay/frontend/data/set/admin/web/internal/portlet/action/EditDataSetMVCRenderCommand.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.frontend.data.set.admin.web.internal.portlet.action;
+
+import com.liferay.frontend.data.set.admin.web.internal.constants.FDSAdminPortletKeys;
+import com.liferay.object.exception.NoSuchObjectDefinitionException;
+import com.liferay.object.exception.NoSuchObjectEntryException;
+import com.liferay.object.model.ObjectDefinition;
+import com.liferay.object.model.ObjectEntry;
+import com.liferay.object.service.ObjectDefinitionLocalService;
+import com.liferay.object.service.ObjectEntryLocalService;
+import com.liferay.object.service.ObjectEntryService;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCRenderCommand;
+import com.liferay.portal.kernel.security.auth.PrincipalException;
+import com.liferay.portal.kernel.security.permission.ActionKeys;
+import com.liferay.portal.kernel.security.permission.PermissionChecker;
+import com.liferay.portal.kernel.security.permission.resource.ModelResourcePermission;
+import com.liferay.portal.kernel.servlet.SessionErrors;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import javax.portlet.PortletException;
+import javax.portlet.RenderRequest;
+import javax.portlet.RenderResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Daniel Sanz
+ */
+@Component(
+	property = {
+		"javax.portlet.name=" + FDSAdminPortletKeys.FDS_ADMIN,
+		"mvc.command.name=/frontend_data_set_admin/edit_data_set"
+	},
+	service = MVCRenderCommand.class
+)
+public class EditDataSetMVCRenderCommand implements MVCRenderCommand {
+
+	@Override
+	public String render(
+			RenderRequest renderRequest, RenderResponse renderResponse)
+		throws PortletException {
+
+		try {
+			String dataSetERC = ParamUtil.getString(
+				_portal.getHttpServletRequest(renderRequest), "dataSetERC");
+
+			ObjectDefinition objectDefinition =
+				_objectDefinitionLocalService.
+					getObjectDefinitionByExternalReferenceCode(
+						"L_DATA_SET", _portal.getCompanyId(renderRequest));
+
+			if (objectDefinition == null) {
+				throw new NoSuchObjectDefinitionException();
+			}
```

Comment:
There's a difference between using `get*` and `fetch*` family of methods:

- `get*` will throw `NoSuchObjectDefinitionException` if definition can't be found. So, no need to have lines 61-63 here.
- `fetch*` will not throw `NoSuchObjectDefinitionException` but instead return `null`. 

For the case, probably we want to just redirect to the data sets page so `get*`  method is ok

About using `ERC` vs `objectDefinitionName` as long as both are unique identifiers, we're fine. Implementations at persistence level are similar. In the case of render action we got here with the ERC so that's the one we'll use.

Getting the `companyId` via `portal` or via `themeDisplay` shall be the [same](https://github.com/liferay/liferay-portal/blob/8f70e6ccfff687801fe0eb95c939d263bdeed643/portal-impl/src/com/liferay/portal/events/ServicePreAction.java#L807) as themeDisplay gets it from portal as well